sdk/batch/azure-batch/azure/batch/custom/custom_errors.py

```diff
@@ -8,12 +8,30 @@ class CreateTasksErrorException(Exception):
     """ Aggregate Exception containing details for any failures from a task add operation.
 
     :param str message: Error message describing exit reason
-    :param [~TaskAddParameter] pending_task_list: List of tasks remaining to be submitted.
+    :param [~TaskAddParameter] pending_tasks: List of tasks remaining to be submitted.
     :param [~TaskAddResult] failure_tasks: List of tasks which failed to add
     :param [~Exception] errors: List of unknown errors forcing early termination
     """
-    def __init__(self, message, pending_task_list=None, failure_tasks=None, errors=None):
-        self.message = message
-        self.pending_tasks = list(pending_task_list)
+    def __init__(self, pending_tasks=None, failure_tasks=None, errors=None):
+        if failure_tasks and errors:
```

Comment:
The way the code is structured here seems a bit confusing to me. If `failure_tasks` and `errors` are both specified, it's assumed that there is more than 1 error and you write the `message` accordingly. But then you immediately drop into `if errors` below and if errors is just 1, you overwrite the message. Then if `failure_tasks` is also jsut one, you overwrite the message again.

It feels like these should be `if`  + `elif` rather than just a bunch of `ifs`

Reply:
yes